keras_cv/src/models/object_detection/faster_rcnn/faster_rcnn.py

```diff
@@ -0,0 +1,653 @@
+import tree
+
+from keras_cv.src import losses
+from keras_cv.src.api_export import keras_cv_export
+from keras_cv.src.backend import keras
+from keras_cv.src.backend import ops
+from keras_cv.src.bounding_box import convert_format
+from keras_cv.src.bounding_box.converters import _decode_deltas_to_boxes
+from keras_cv.src.bounding_box.utils import _clip_boxes
+from keras_cv.src.layers.object_detection.anchor_generator import (
+    AnchorGenerator,
+)
+from keras_cv.src.layers.object_detection.box_matcher import BoxMatcher
+from keras_cv.src.layers.object_detection.non_max_suppression import (
+    NonMaxSuppression,
+)
+from keras_cv.src.layers.object_detection.roi_align import ROIAligner
+from keras_cv.src.layers.object_detection.roi_generator import ROIGenerator
+from keras_cv.src.layers.object_detection.roi_sampler import ROISampler
+from keras_cv.src.layers.object_detection.rpn_label_encoder import (
+    RpnLabelEncoder,
+)
+from keras_cv.src.models.object_detection.__internal__ import unpack_input
+from keras_cv.src.models.object_detection.faster_rcnn import FeaturePyramid
+from keras_cv.src.models.object_detection.faster_rcnn import RCNNHead
+from keras_cv.src.models.object_detection.faster_rcnn import RPNHead
+from keras_cv.src.models.task import Task
+from keras_cv.src.utils.train import get_feature_extractor
+
+BOX_VARIANCE = [0.1, 0.1, 0.2, 0.2]
+
+
+@keras_cv_export(
+    [
+        "keras_cv.models.FasterRCNN",
+        "keras_cv.models.object_detection.FasterRCNN",
+    ]
+)
+class FasterRCNN(Task):
+    def __init__(
+        self,
+        backbone,
+        num_classes,
+        bounding_box_format,
+        anchor_generator=None,
+        feature_pyramid=None,
+        fpn_min_level=2,
+        fpn_max_level=5,
+        rpn_head=None,
+        rpn_filters=256,
+        rpn_kernel_size=3,
+        rpn_label_en_pos_th=0.7,
```

Comment:
lets abbreviate the args completely
`rpn_label_en_pos_th`-> `rpn_label_encoder_positive_threshold`. Same for all the args.

Reply:
Done